plugins/Serial/__init__.py

```diff
@@ -111,14 +72,42 @@ def MyHexDecoder(input):
 ]
 
 
+def RevokeWorkPermit(self):
+    if self.plugin.ReceiveThread is not None:
+        if self.plugin.ReceiveThread_WorkPermit.is_set() is True:
+            self.plugin.ReceiveThread_WorkPermit.clear()
+            time.sleep(self.plugin.SerialPortInterface.timeout)
+        else:
+            print("The worker thread is already paused.")
+    else:
+        print("'Generate events' is not enabled, so the worker thread does not need to be paused.")
+
+
+def GrantWorkPermit(self):
+    if self.plugin.ReceiveThread is not None:
+        if self.plugin.ReceiveThread_WorkPermit.is_set() is False:
+            FlushInput(self)
+            self.plugin.ReceiveThread_WorkPermit.set()
+        else:
+            print("The worker thread is already running.")
+    else:
+        print("'Generate events' is not enabled, so the worker thread can not be unpaused.")
+
+
+def FlushInput(self):
+    self.plugin.SerialPortInterface.flushInput()
+
+
 class Serial(eg.RawReceiverPlugin):
-    text = Text
 
     def __init__(self):
         eg.RawReceiverPlugin.__init__(self)
-        self.AddAction(Write)
-        self.AddAction(Read)
-        self.serial = None
+        self.AddAction(fnWrite)
+        self.AddAction(fnRead)
+        self.AddAction(fnPauseEventGeneration)
+        self.AddAction(fnFlushInput)
+        self.AddAction(fnContinueEventGeneration)
+        self.SerialPortInterface = None
```

Comment:
the use of the fn in the last example we typically only use when generating the actions from a list. we do this as an identifier to let us know that the class is dynamically created for the action. it is not something that is hard coded. if it is a hard coded class stick to the PEP8  camel case naming convention.